### DISCUSSION TITLE
CreateItems gets a 204 if API is Write-Only, how can I catch this

### DISCUSSION THREAD

**Discussion by jonock** [Intevel/nuxt-directus#251] [Q&A]
Hey Everyone

I'm using Directus to collect answers to a form in my nuxt application. 
Since I don't want to expose all form answers via the API, I only allow write access to the public. 

![image](https://github.com/Intevel/nuxt-directus/assets/4542186/6048d7c0-b21c-4ac4-9779-31ed3f158e39)

Now comes the problem: Directus creates the entry perfectly but answers with a 204 if reading is not permitted from the API. This means, Nuxt throws `TypeError: items is undefined`. 

Can you help me how to catch this properly without rewriting the function itself?

**Comment by jonock** [comment]
My workaround for now is to just catch the error when calling the function but that's not really neat from my perspective.

**Reply by Intevel** [reply]
What do you think would be better?

**Reply by jonock** [reply]
I think since a 204 seems to be somewhat the correct and intended behaviour of Directus. The createItems function should catch this and not expect a response with an object.

**Comment by sandros94** [comment]
Are you using `nuxt-directus@5.x` or [`nuxt-directus-next`](https://github.com/Intevel/nuxt-directus/discussions/215)?

**Reply by jonock** [reply]
I'm currently using `nuxt-directus@5.x`
